### PR TITLE
Port Handle EOF HResult from async callback to 2.2

### PR DESCRIPTION
**Pull request by pakrym**
Fixes https://github.com/aspnet/AspNetCore/issues/6415

### Description

When using in-proc mode with ANCM, POST requests can randomly result in `contex.Request.Body.ReadAsync` throwing because successful error code is not handled correctly in all places.

### Regression?
Regression from out-of-process hosting model.

## Risk

Very Low. Handling same error code we did before but in more centralized fashion.

### CODE REVIEW

**Comment by pakrym** [comment]
cc @muratg

**Review by jkotalik — APPROVED** [review]

**Comment by natemcmaster** [comment]
Please wait until https://github.com/aspnet/AspNetCore-Internal/issues/1610 is resolved before merging.

**Comment by jamshedd** [comment]
Approved for 2.2.2.

File: eng/PatchConfig.props [hunk 1]
```diff
@@ -27,9 +27,10 @@ Later on, this will be checked using this condition:
   </PropertyGroup>
   <PropertyGroup Condition=" '$(VersionPrefix)' == '2.2.2' ">
     <PackagesInPatch>
-      @aspnet/signalr;
```
Comment (natemcmaster) [inline eng/PatchConfig.props]: FYI @BrennanConroy  

Intentional removal? We need this if we want to patch NPM packages.

Comment (BrennanConroy) [reply eng/PatchConfig.props]: 😢

Comment (pakrym) [reply eng/PatchConfig.props]: Bad merge.

Comment (muratg) [reply eng/PatchConfig.props]: @pakrym You're fixing it?

**Pull request merged**